base/applications/rapps/loaddlg.cpp

```diff
@@ -430,732 +432,736 @@ MessageBox_LoadString(HWND hOwnerWnd, INT StringID)
     }
 }
 
+static BOOL
+ShowLastError(HWND hWndOwner, BOOL bInetError, DWORD dwLastError)
+{
+    CLocalPtr<WCHAR> lpMsg;
+
+    if (!FormatMessageW(
+            FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_IGNORE_INSERTS |
+                (bInetError ? FORMAT_MESSAGE_FROM_HMODULE : FORMAT_MESSAGE_FROM_SYSTEM),
+            (bInetError ? GetModuleHandleW(L"wininet.dll") : NULL), dwLastError, LANG_USER_DEFAULT, (LPWSTR)&lpMsg, 0,
+            NULL))
+    {
+        DPRINT1("FormatMessageW unexpected failure (err %d)\n", GetLastError());
+        return FALSE;
+    }
+
+    MessageBoxW(hWndOwner, lpMsg, NULL, MB_OK | MB_ICONERROR);
+    return TRUE;
+}
+
 // Download dialog (loaddlg.cpp)
-class CDownloadManager
+HWND g_hDownloadWnd = NULL;
```

Comment:
Or make it a public static member of the CDownloadManager class (accessible via `CDownloadManager::hWnd`) ?